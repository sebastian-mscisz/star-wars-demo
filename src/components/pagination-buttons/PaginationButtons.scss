@import 'src/variables';

.sw-pagination {
    &-section {
        width: 100%;
        margin-top: 30px;
        margin-right: 20px;
        display: inline-flex;
        justify-content: flex-end;

        &--center {
            justify-content: center;
        }
    }

    &-button {
        font-size: x-large;
        color: $base-color;
        width: 50px;
        height: 50px;
        background-color: $base-background-color;
        border: 2px solid $base-color;
        margin: 5px;
        border-radius: 15%;
        display: flex;
        justify-content: center;
        align-items: center;
        cursor: pointer;

        &--hidden {
            visibility: hidden;
        }

        &:hover {
            border-color: $base-color-light;
            color: $base-color-light;
        }
    }
}
