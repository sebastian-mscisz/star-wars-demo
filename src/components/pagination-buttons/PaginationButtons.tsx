import { useNavigate } from 'react-router-dom';
import './PaginationButtons.scss';

type PaginationButtonsProps = {
    hideLeft?: boolean;
    hideRight?: boolean;
    center?: boolean;
    leftPath: string;
    rightPath: string;
};

export const PaginationButtons: React.FC<PaginationButtonsProps> = ({
    hideLeft,
    hideRight,
    leftPath,
    rightPath,
    center,
}) => {
    const navigate = useNavigate();

    return (
        <div
            data-testid="pagination-section"
            className={`sw-pagination-section${center ? ' sw-pagination-section--center' : ''}`}
        >
            <div
                data-testid="left-pagination-button"
                onClick={() => {
                    navigate(leftPath);
                }}
                className={`sw-pagination-button ${hideLeft ? 'sw-pagination-button--hidden' : ''}`}
            >
                &#8592;
            </div>
            <div
                data-testid="right-pagination-button"
                onClick={() => {
                    navigate(rightPath);
                }}
                className={`sw-pagination-button ${hideRight ? 'sw-pagination-button--hidden' : ''}`}
            >
                &#8594;
            </div>
        </div>
    );
};
