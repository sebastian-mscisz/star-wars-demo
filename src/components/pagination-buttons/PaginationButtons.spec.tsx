import { BrowserRouter } from 'react-router-dom';
import { PaginationButtons } from './PaginationButtons';
import { screen } from '@testing-library/dom';
import { render } from '@testing-library/react';

describe('Pagination buttons', () => {
    let leftButton: HTMLElement;
    let rightButton: HTMLElement;

    const getButtons = () => {
        leftButton = screen.getByTestId('left-pagination-button');
        rightButton = screen.getByTestId('right-pagination-button');
    };

    it('should render', () => {
        const { asFragment } = render(<PaginationButtons leftPath="/left" rightPath="/right" />, {
            wrapper: BrowserRouter,
        });
        expect(asFragment()).toMatchSnapshot();
    });

    it('with hidden left button', () => {
        render(<PaginationButtons leftPath="/left" rightPath="/right" hideLeft />, { wrapper: BrowserRouter });
        getButtons();
        expect(leftButton.classList.contains('sw-pagination-button--hidden')).toBe(true);
        expect(rightButton.classList.contains('sw-pagination-button--hidden')).toBe(false);
    });

    it('with hidden right button', () => {
        render(<PaginationButtons leftPath="/left" rightPath="/right" hideRight />, { wrapper: BrowserRouter });
        getButtons();
        expect(leftButton.classList.contains('sw-pagination-button--hidden')).toBe(false);
        expect(rightButton.classList.contains('sw-pagination-button--hidden')).toBe(true);
    });

    it('with hidden both buttons', () => {
        render(<PaginationButtons leftPath="/left" rightPath="/right" hideLeft hideRight />, {
            wrapper: BrowserRouter,
        });
        getButtons();
        expect(leftButton.classList.contains('sw-pagination-button--hidden')).toBe(true);
        expect(rightButton.classList.contains('sw-pagination-button--hidden')).toBe(true);
    });

    it('with centered buttons', () => {
        render(<PaginationButtons leftPath="/left" rightPath="/right" center />, { wrapper: BrowserRouter });
        const paginationSection = screen.getByTestId('pagination-section');
        expect(paginationSection.classList.contains('sw-pagination-section--center')).toBe(true);
    });
});
