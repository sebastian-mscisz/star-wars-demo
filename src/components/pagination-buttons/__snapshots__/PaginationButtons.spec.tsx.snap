// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination buttons should render 1`] = `
<DocumentFragment>
  <div
    class="sw-pagination-section"
    data-testid="pagination-section"
  >
    <div
      class="sw-pagination-button "
      data-testid="left-pagination-button"
    >
      ←
    </div>
    <div
      class="sw-pagination-button "
      data-testid="right-pagination-button"
    >
      →
    </div>
  </div>
</DocumentFragment>
`;
